### DISCUSSION TITLE
Is there an upgrade path for old installations?

### DISCUSSION THREAD

**Discussion by sbabcoc** [TermuxArch/TermuxArch#92] [Q&A] [answered]
@SDRausty Thank you for providing this incredible project! I wasn't aware of the `update` feature and am apparently way behind current state. The installation hierarchy doesn't match what the `update` feature expects to find, and no update is performed. Is there any way to rescue my current installation, or is it best to just blow away my current installation and start over?
I have an entire development environment set up in Arch Linux on my phone, and I'd rather not have to set everything up again, but the current installation has more than a few thinks that don't behave properly, and my `mozilla-central` project build keeps crashing.

**Comment by SDRausty** [comment]
You are very welcome.  Many people are making this project come true, and thank you for touching on an important topic:
>hierarchy doesn't match what the `update` feature expects to find, and no update is performed. 

There are more than two `update` functions.  The `refresh` option is being discussed here.  Commit 
https://github.com/TermuxArch/TermuxArch/commit/b2938eca7ba9f248ad3f7a92ead335476f20800e might resolve the following notes once it is fully implemented:

>expects to find, and no update is performed. 

You can disable this feature with these commands:

`setupTermuxArch bloom`
`cd ~/TermuxArchBloom`

Commenting out function [\_DODIRCHK_ in line 352 in file `setupTermuxArch`](https://github.com/TermuxArch/TermuxArch/blob/master/setupTermuxArch#L352) should resolve 'no update is performed'.

>any way to rescue my current installation

This dicussion is limited to being able to refreshing the TermuxArch files in order to resolve "Is there an upgrade path for old installations?" easily.  I have no idea whether you can rescue your current installation. Good luck:

>start over?

With the newly implemented cache feature, backing up your work and starting over might be easier.  Please share your progress here.  Happy Easter 🐰

**Comment by SDRausty** [comment]
Commit https://github.com/TermuxArch/TermuxArch/commit/b6eb0fc1dfd688ae9fc191b154e14a70e8280bb6 implements a working version of  overriding the hierarchy match.  The command `setupTermuxArch refreshforce` should resolve 'updgrade path for old installation';  Can you please see whether the `refreshforce` option performs as expected on your installation @sbabcoc?

**Reply by sbabcoc** [reply]
I discovered a much more fundamental issue that I think recommends the "scrap it and start over" approach. The version of Termux on my device is really old, (0.101) and simply installing the latest APK does nothing.

**Comment by SDRausty** [comment, marked as answer]
>version of Termux on my device is really old, (0.101) and simply installing the latest APK does nothing.

I ran into a similar bump a while ago (Termux app would not upgrade);  Termux has begun releasing APKs via Github at https://github.com/termux/termux-app/releases that you can install directly from Github.

Termux app APK debug releases are available at https://github.com/termux/termux-app/actions/workflows/debug_build.yml

**Comment by JanuszChmiel** [comment]
Because AArch64 repository still contain non functioning version of Seamonkey, I will do The following:
I will run makeyay script to install aurhelpers.
I will type
yay install seamonkey
And I will try to build this complex bundle. I will report errors which are preventing Me from building Seamonkey. 
If Seamonkey ArchLinux repository contain nonfunctional package even for X86 which uses regular Arch Linux full installation, it will be necessary to build slightly older Seamonkey version for AArch 64. Then it would be necessary to host AArch64 architecture package outside of official Aur repository to allow us to use Seamonkey with our Android devices.

**Comment by JanuszChmiel** [comment]
Oh. It is necessary to type yay -S seamonkey.
But this do not work, because Pacman find non functioning available Seamonkey on Aarch64 community repository so yay can not build Seamonkey from scratch.
